### PR TITLE
Only check voltage invoice is within 10 sats

### LINKED ISSUE #1046: Received unexpected invoice from LSP
```
af04 2024-02-22 14:06:24.228 ERROR [mutiny_core::node:1159] Received unexpected invoice from LSP: Some(195630815) when amount was 195630000
```

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #1046

### CODE REVIEW

File: mutiny-core/src/node.rs [hunk 1]
```diff
@@ -1158,7 +1158,8 @@ impl<S: MutinyStorage> Node<S> {
 
                         if lsp_invoice.payment_hash() != invoice.payment_hash()
                             || lsp_invoice.recover_payee_pub_key() != lsp_pubkey
-                            || (lsp_invoice.amount_milli_satoshis() != Some(amount_sat * 1_000)
+                            || (lsp_invoice.amount_milli_satoshis().map(|s| s / 1_000)
```
Comment (TonyGiorgio) [inline mutiny-core/src/node.rs:1161]: is there not just an `amount_sats` ? What if it rounds the wrong way?

Comment (benthecarman) [reply mutiny-core/src/node.rs:1161]: rust always rounds down, ldk only has a msats option

Comment (benthecarman) [reply mutiny-core/src/node.rs]: changed to just check within 10 sats

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
I think this should be pulled out to a different method and tested

**Review by TonyGiorgio — APPROVED** [review]

**Pull request merged**